lib/elixir/test/elixir/kernel_test.exs

```diff
@@ -301,23 +301,39 @@ defmodule KernelTest do
   end
 
   test "defdelegate/2 raises with non-variable arguments" do
-    code = quote do
-      defmodule Foo do
-        defdelegate foo(1), to: List
-      end
-    end
+    msg = "defdelegate/2 only accepts function parameters, got: 1"
 
-    msg = "defdelegate/2 only accepts variable names, got: 1"
-    assert_raise ArgumentError, msg, fn -> Code.compile_quoted(code) end
+    assert_raise ArgumentError, msg, fn -> Code.eval_string("""
+        defmodule Foo do
+          defdelegate foo(1), to: List
+        end
+      """, [], __ENV__)
+    end
 
-    code = quote do
-      defmodule Foo do
-        defdelegate foo(a \\ 1), to: List
-      end
+    assert_raise ArgumentError, msg, fn -> Code.eval_string("""
+        defmodule FooBar do
+          defdelegate foo(1 \\\\ 1), to: List
+        end
+      """, [], __ENV__)
     end
+  end
+
+  defdelegate my_reverse(list \\ []), to: :lists, as: :reverse
```

Comment:
I just tried to use `Code.eval_string`  but it got pretty ugly! I can see the point for code raising errors, but I do not think it is a very good idea here, as real functions will give a more readable output on failure.